runtime/marshal_jsonpb.go

```diff
@@ -218,6 +246,23 @@ func decodeNonProtoField(d *json.Decoder, v interface{}) error {
 		}
 		return nil
 	}
+	if rv.Kind() == reflect.Slice {
+		var sl []json.RawMessage
+		if err := d.Decode(&sl); err != nil {
+			return err
+		}
+		if sl != nil {
+			rv.Set(reflect.MakeSlice(rv.Type(), 0, 0))
+		}
+		for _, item := range sl {
+			bv := reflect.New(rv.Type().Elem())
+			if err := unmarshalJSONPb([]byte(item), bv.Interface()); err != nil {
+				return err
+			}
+			rv.Set(reflect.Append(rv, bv.Elem()))
+		}
+		return nil
+	}
```

Comment:
I had to add this to support marshalling slices of proto messages, not sure how this worked before.